src/Tests/dotnet-new.Tests/CommonTemplatesTests.cs

```diff
@@ -199,6 +202,56 @@ public void AllCommonItemsCreate(string expectedTemplateName, string templateSho
             Directory.Delete(workingDir, true);
         }
 
+        private static void InstanciateAndAssert(ITestOutputHelper log, SharedHomeDirectory root, string templateName, string templateShortName, string[] templateArgs, Action<string> assertGivenWorkingDirectory)
+        {
+            string workingDir = TestUtils.CreateTemporaryFolder();
+            var allArgs = new List<string>(1 + templateArgs.Length)
+            {
+                templateShortName
+            };
+            allArgs.AddRange(templateArgs);
+            new DotnetNewCommand(log, allArgs.ToArray())
+                .WithCustomHive(root.HomeDirectory)
+                .WithWorkingDirectory(workingDir)
+                .Execute()
+                .Should()
+                .ExitWith(0)
+                .And.NotHaveStdErr()
+                .And.HaveStdOut($@"The template ""{templateName}"" was created successfully.");
+            assertGivenWorkingDirectory(workingDir);
+            Directory.Delete(workingDir, true);
+        }
+
+        private static void HasImport(string filePath) => File.ReadAllText(filePath).Should().Contain("<Import");
+
+        [Fact]
+        public void BuildPropsTemplateCalculatesInclude() =>
+            InstanciateAndAssert(
+                _log,
+                _fixture,
+                "MSBuild Directory.Build.props file",
+                "buildprops",
+                new[]
+                {
+                    "--inherit"
+                },
+                workingDir => HasImport(Path.Combine(workingDir, "Directory.Build.props"))
+            );
+
+        [Fact]
+        public void BuildTargetsTemplateCalculatesInclude() =>
+            InstanciateAndAssert(
+                _log,
+                _fixture,
+                "MSBuild Directory.Build.targets file",
+                "buildtargets",
+                new[]
+                {
+                    "--inherit"
+                },
+                workingDir => HasImport(Path.Combine(workingDir, "Directory.Build.targets"))
+            );
+
```

Comment:
Those tests can be easier and more complete as snapshot tests - they will verify the whole content then.

Reply:
Done, thanks for the pointer